### PR TITLE
Feat!: add more interval units and make the smallest 5 minutes

### PR DESCRIPTION
**Pull request by tobymao**
additional performance boosts

still need to add some tests and validate that this doesn't break existing minute level models

### CODE REVIEW

File: sqlmesh/core/plan/definition.py [hunk 1]
```diff
@@ -783,7 +783,9 @@ def is_finished(self) -> bool:
         return self == PlanStatus.FINISHED
 
 
-class SnapshotIntervals(PydanticModel, frozen=True):
+# millions of these can be created, pydantic has significant overhead
```
Comment (izeigerman) [inline sqlmesh/core/plan/definition.py:786]: Where do millions come from? Isn't it just one per snapshot?

File: sqlmesh/core/plan/definition.py [hunk 2]
```diff
@@ -795,11 +797,12 @@ def format_intervals(self, unit: t.Optional[IntervalUnit] = None) -> str:
         return format_intervals(self.merged_intervals, unit)
 
 
+@dataclass
```
Comment (izeigerman) [inline sqlmesh/core/plan/definition.py:800]: Ditto

File: sqlmesh/core/node.py [hunk 3]
```diff
@@ -33,17 +36,19 @@ class IntervalUnit(str, Enum):
...
+    def from_cron(klass, cron: str, sample_size: int = DEFAULT_SAMPLE_SIZE) -> IntervalUnit:
         croniter = CroniterCache(cron)
         samples = [croniter.get_next() for _ in range(sample_size)]
         min_interval = min(b - a for a, b in zip(samples, samples[1:]))
         for unit, seconds in sorted(INTERVAL_SECONDS.items(), key=lambda x: x[1], reverse=True):
-            if seconds <= min_interval:
+            if seconds <= min_interval.total_seconds():
                 return unit
-        raise ConfigError(f"Invalid cron '{cron}': must have a cadence of 1 minute or more.")
+        raise ConfigError(f"Invalid cron '{cron}': must have a cadence of 5 minutes or more.")
```
Comment (izeigerman) [inline sqlmesh/core/node.py:51]: Do we prohibit using `minute` grain anywhere?

File: sqlmesh/core/plan/definition.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/plan/definition.py:786]: i gues intervals can be a lot and then pydantic validates it. changing to data class reduces plan by 5x in my tests

File: sqlmesh/core/node.py (hunk 3 above)
Comment (tobymao) [reply sqlmesh/core/node.py:51]: yea, it's not an enum anymore

**Review by izeigerman — APPROVED** [review]

**Pull request merged**